### PR TITLE
Model tumor factors

### PR DESCRIPTION
**Pull request by MariekeSpanFurore**

### CODE REVIEW

**Comment by vadi2** [comment]
We had a look at this offline - what's the latest info on this @MariekeSpanFurore?

**Comment by MariekeSpanFurore** [comment]
I still have these errors, so I'll ignore them as long as you don't have them as well. I cannot find the problem.

**Comment by MariekeSpanFurore** [comment]
I have added the new elements from the V5 version of the BRC set. They are now missing codes because these were not mapped and I also don't have any examples. So this is not ready to publish, but with what I have now there finished.

File: input/fsh/Terminology.fsh [hunk 1]
```diff
@@ -83,6 +84,24 @@ Description: "Valueset of the grade of the tumor"
 * SCT#399611001 "High grade histologic differentiation (finding)"
 * NullFlavor#NI "No information"
 
+ValueSet: TNMPrimaryTumorVS
+Id: mcode-tnm-primary-tumor-category-vs
+Title: "TNM Primary Tumor Category Value Set"
+Description: "Valueset of the TNM stage for the T category, according to TNM staging rules"
+* codes from system AJCC
```
Comment (vadi2) [inline input/fsh/Terminology.fsh:91]: Can you explicitly enumerate (list out) the AJCC codes here, just like we do with snomed above for example?

File: input/fsh/Profiles clinical datapoints/TumorFactors.fsh [hunk 2]
```diff
@@ -0,0 +1,394 @@
...
+Id: mammaprint-score
+Title: "Mammaprint score"
+Description: "Represents the mammaprintscore on a scale of 0.000 to 1.000"
+* insert PublicationProfileRuleset
+* subject only Reference(BreastCancerPatient)
+* value[x] only Quantity
+* value[x] MS
+* effectiveDateTime MS
+// * method --> we would need a code for this
+// can we limit the scale to 0.000 to 1.000? And do we want that?
```
Comment (vadi2) [inline input/fsh/Profiles clinical datapoints/TumorFactors.fsh:287]: We actually can, looks like there is minValue and maxValue available in the FHIR profile - https://www.hl7.org/fhir/elementdefinition-definitions.html#ElementDefinition.minValue_x_

Just need to work out how to set it via FSH and test that it actually works in a validator.

File: input/fsh/Profiles clinical datapoints/TumorStaging.fsh [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+Description: "Represents the pathological stage of the tumor (per AJCC 8th Ed.)"
+* insert PublicationProfileRuleset
+* code = SCT#385356007 "Tumor stage finding"
+* subject only Reference(BreastCancerPatient)
+* value[x] only CodeableConcept 
+* value[x] from TNMPrimaryTumorVS (preferred)
+* value[x] MS
+* effectiveDateTime MS
+
+Instance: PathologicalTNMTumorStagePatient147
```
Comment (RicardoQuintano) [inline input/fsh/Profiles clinical datapoints/TumorStaging.fsh:45]: Shouldn't it also include the code and the effectiveDateTime? In my opinion it is important to provide complete examples.

code = SCT#385356007 "Tumor stage finding"

File: input/fsh/Profiles clinical datapoints/TumorStaging.fsh [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+Title: "Pathological TNM tumor stage to ICHOM set"
+Description: "Mapping of the pathological TNM tumor stage to the ICHOM breast cancer PCOM set." 	
+* valueCodeableConcept -> "Pathological tumor stage"
+
+
+// CLINICAL NODAL STAGE
+Profile: ClinicalTNMNodalStage
+Id: clinical-tnm-nodal-stage
+Parent: Observation
+Title: "Clinical TNM nodal stage"
```
Comment (RicardoQuintano) [inline input/fsh/Profiles clinical datapoints/TumorStaging.fsh:67]: Suggestion: include full description of TNM

File: input/fsh/Profiles clinical datapoints/TumorStaging.fsh [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+Description: "Represents the clinical stage of the presence or absence of metastases in regional lymph nodes (per AJCC 8th Ed.)"
+* insert PublicationProfileRuleset
+* code = SCT#385382003 "nodal stage finding"
+* subject only Reference(BreastCancerPatient)
+* value[x] only CodeableConcept 
+* value[x] from TNMRegionalNodesVS (preferred)
+* value[x] MS
+* effectiveDateTime MS
+
+Instance: ClinicalTNMNodalStagePatient147
```
Comment (RicardoQuintano) [inline input/fsh/Profiles clinical datapoints/TumorStaging.fsh:77]: Shouldn't it also include the code and the effectiveDateTime? In my opinion it is important to provide complete examples.

* code = SCT#385382003 "nodal stage finding"

File: input/fsh/Profiles clinical datapoints/TumorStaging.fsh [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+Title: "Clinical TNM nodal stage to ICHOM set"
+Description: "Mapping of the clinical TNM nodal stage to the ICHOM breast cancer PCOM set." 	
+* valueCodeableConcept -> "Clinical nodal stage"
+
+
+// PATHOLOGICAL NODAL STAGE
+Profile: PathologicalTNMNodalStage
+Id: pathological-tnm-nodal-stage
+Parent: Observation
+Title: "Pathological TNM nodal stage"
```
Comment (RicardoQuintano) [inline input/fsh/Profiles clinical datapoints/TumorStaging.fsh:98]: Suggestion: include full description of TNM

File: input/fsh/Profiles clinical datapoints/TumorStaging.fsh [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+Source:	ClinicalTNMNodalStage
+Target: "https://connect.ichom.org/patient-centered-outcome-measures/breast-cancer"
+Id: ClinicalTNMNodalStageMapping
+Title: "Clinical TNM nodal stage to ICHOM set"
+Description: "Mapping of the clinical TNM nodal stage to the ICHOM breast cancer PCOM set." 	
+* valueCodeableConcept -> "Clinical nodal stage"
+
+
+// PATHOLOGICAL NODAL STAGE
+Profile: PathologicalTNMNodalStage
```
Comment (RicardoQuintano) [inline input/fsh/Profiles clinical datapoints/TumorStaging.fsh:95]: This one looks a copy of ClinicalTNMNodalStage profile. Could we have only one profile?

File: input/fsh/Profiles clinical datapoints/TumorStaging.fsh [hunk 8]
```diff
@@ -0,0 +1,184 @@
...
+Description: "Mapping of the pathological TNM nodal stage to the ICHOM breast cancer PCOM set." 	
+* valueCodeableConcept -> "Pathological nodal stage"
+
+
+
+// CLINICAL DISTANT METASTASES STAGE
+Profile:  ClinicalTNMDistantMetastases
+Id: clinical-tnm-distant-metastases
+Parent: Observation
+Title: "Clinical TNM distant metastases"
```
Comment (RicardoQuintano) [inline input/fsh/Profiles clinical datapoints/TumorStaging.fsh:130]: Suggestion: include full description of TNM

File: input/fsh/Profiles clinical datapoints/TumorStaging.fsh [hunk 9]
```diff
@@ -0,0 +1,184 @@
...
+Description: "Represents the clinical stage of the extent of a tumor metastasis in remote anatomical locations (per AJCC 8th Ed.)"
+* insert PublicationProfileRuleset
+* code = SCT#385380006 "Metastasis category finding"
+* subject only Reference(BreastCancerPatient)
+* value[x] only CodeableConcept 
+* value[x] from TNMDistantMetastasesVS (preferred)
+* value[x] MS
+* effectiveDateTime MS
+
+Instance: ClinicalTNMDistantMetastasesPatient147
```
Comment (RicardoQuintano) [inline input/fsh/Profiles clinical datapoints/TumorStaging.fsh:140]: Shouldn't it also include the code and the effectiveDateTime? In my opinion it is important to provide complete examples.

* code = SCT#385380006 "Metastasis category finding"

File: input/fsh/Profiles clinical datapoints/TumorStaging.fsh [hunk 10]
```diff
@@ -0,0 +1,184 @@
...
+Mapping: ClinicalTNMDistantMetastasesToICHOM
+Source:	ClinicalTNMDistantMetastases
+Target: "https://connect.ichom.org/patient-centered-outcome-measures/breast-cancer"
+Id: ClinicalTNMDistantMetastasesMapping
+Title: "TNM score of clinical distant metastases to ICHOM set"
+Description: "Mapping of the TNM score of the clinical distant metastases to the ICHOM breast cancer PCOM set." 	
+* valueCodeableConcept -> "Clinical distant metastasis"
+
+// PATHOLOGICAL DISTANT METASTASES STAGE
+Profile:  PathologicalDistantMetastasesTNM
```
Comment (RicardoQuintano) [inline input/fsh/Profiles clinical datapoints/TumorStaging.fsh:157]: This one looks a copy of ClinicalTNMDistantMetastases profile. Could we have only one profile?

File: input/fsh/Profiles clinical datapoints/TumorStaging.fsh [hunk 11]
```diff
@@ -0,0 +1,184 @@
...
+Id: pathological-tnm-distant-metastases
+Parent: Observation
+Title: "Pathological TNM distant metastases"
+Description: "Represents the pathological stage of the extent of a tumor metastasis in remote anatomical locations (per AJCC 8th Ed.)"
```
Comment (RicardoQuintano) [inline input/fsh/Profiles clinical datapoints/TumorStaging.fsh:161]: Suggestion: include full description of TNM

File: input/fsh/Profiles clinical datapoints/TumorStaging.fsh [hunk 12]
```diff
@@ -0,0 +1,184 @@
...
+* insert PublicationProfileRuleset
+* code = SCT#385380006 "Metastasis category finding"
+* subject only Reference(BreastCancerPatient)
+* value[x] only CodeableConcept 
+* value[x] from TNMDistantMetastasesVS (preferred)
+* value[x] MS
+* effectiveDateTime MS
+
+Instance: PathologicalDistantMetastasesTNMPatient147
```
Comment (RicardoQuintano) [inline input/fsh/Profiles clinical datapoints/TumorStaging.fsh:170]: Shouldn't it also include the code and the effectiveDateTime? In my opinion it is important to provide complete examples.

* code = SCT#385380006 "Metastasis category finding"

File: input/fsh/Profiles clinical datapoints/TumorFactors.fsh [hunk 13]
```diff
@@ -0,0 +1,394 @@
...
+
+Mapping: TumorGradeToICHOM
+Source:	TumorGrade
+Target: "https://connect.ichom.org/patient-centered-outcome-measures/breast-cancer"
+Id: tumorgrademapping
+Title: "Tumor grade to ICHOM set"
+Description: "Mapping of the tumor grade to the ICHOM breast cancer PCOM set" 	
+* valueCodeableConcept ->  "Tumor grade"
+
+//INVASION GRADE
```
[the remark this answers is not in the document]
Comment (MariekeSpanFurore) [reply input/fsh/Profiles clinical datapoints/TumorFactors.fsh:87]: In a previous version, I sliced this observation and included both elements in a single profile. However, these are two different observations, that are not necessarily measured together. So although they are very similar, I think it's best to have them separate.

File: input/fsh/Profiles clinical datapoints/TumorFactors.fsh [hunk 14]
```diff
@@ -0,0 +1,394 @@
...
+* effectiveDateTime MS
+
+Instance: TumorSizePatient147
+InstanceOf: TumorSize
+Description: "Example of the size of the invasive tumor of a patient with breast cancer"
+* status = ObservationStatusCS#final
+* subject = Reference(BreastCancerPatient147)
+* valueQuantity.value = 59
+
+Mapping: TumorSizeToICHOM
```
[the remark this answers is not in the document]
Comment (MariekeSpanFurore) [reply input/fsh/Profiles clinical datapoints/TumorFactors.fsh:136]: I agree

File: input/fsh/Profiles clinical datapoints/TumorFactors.fsh [hunk 15]
```diff
@@ -0,0 +1,394 @@
...
+* insert PublicationProfileRuleset
+* code = SCT#364108009 "Lymph node observable"
+* subject only Reference(BreastCancerPatient)
+* component ^slicing.discriminator.type = #pattern
+* component ^slicing.discriminator.path = "code"
+* component ^slicing.rules = #open
+* component ^slicing.description = "Slice based on the component.code pattern"
+* component contains 
+    LymphNodesResected 0..1 MS and LymphNodesInvolved 0..1 MS
+* component[LymphNodesResected] ^short = "Number lymph nodes resected"
```
[the remark this answers is not in the document]
Comment (MariekeSpanFurore) [reply input/fsh/Profiles clinical datapoints/TumorFactors.fsh:160]: I think your right, I was not entirely sure about this. Both elements could follow from the same pathological review, but they could also be measured separately (e.g. the number of resected lymph nodes could be reported in a surgery report, while the number of the involved lymph nodes follows from a pathological review. I think it's useful to ask ICHOM's opinion on this too.

File: input/fsh/Profiles clinical datapoints/TumorFactors.fsh [hunk 16]
```diff
@@ -0,0 +1,394 @@
...
+
+Mapping: GermlineMutationToICHOM
+Source:	GermlineMutation
+Target: "https://connect.ichom.org/patient-centered-outcome-measures/breast-cancer"
+Id: GermlineMutationMapping
+Title: "Germline mutation to ICHOM set"
+Description: "Mapping of the germline mutation to the ICHOM breast cancer PCOM set" 	
+* valueCodeableConcept -> "Germline mutation"
+
+// TUMOR GRADING
```
[the remark this answers is not in the document]
Comment (MariekeSpanFurore) [reply input/fsh/Profiles clinical datapoints/TumorFactors.fsh:59]: I see that I haven't been very consistent with adding effectiveDateTime as MS. 

For the profile histotype I have an example of the date because the date of histological diagnosis is explicitly in the ICHOM set (as an element). For the other profiles the date is not explicitly in the ICHOM set. However, timing is an important factor in the set, because the data is repeatedly collected during follow-up of the patient. So I figured the timing could be relevant to understand the timing of these observations, that's why I wanted to profile them as MS, but forgot to add them to all the profiles. For the instances I don't have example dates, but I could reuse the date of the histological diagnosis. 

In short: Do we want effectiveDateTime as MS in all the observations? And if yes, do we want example dates as well?

What do you think, @vadi2

File: input/fsh/Profiles clinical datapoints/TumorFactors.fsh [hunk 17]
```diff
@@ -0,0 +1,394 @@
...
+Source:	EndopredictonScore
+Target: "https://connect.ichom.org/patient-centered-outcome-measures/breast-cancer"
+Id: Endopredictonmapping
+Title: "Endopredicton scoreto ICHOM set"
+Description: "Mapping of the Endopredicton score to the ICHOM breast cancer PCOM set" 	
+* valueQuantity -> "Endopredicton score"
+
+
+// Multidisciplinary meeting 
+Profile: MultidisciplinaryMeeting
```
[the remark this answers is not in the document]
Comment (MariekeSpanFurore) [reply input/fsh/Profiles clinical datapoints/TumorFactors.fsh:366]: I agree, I changed the naming.

File: input/fsh/Profiles clinical datapoints/TumorFactors.fsh (hunk 2 above)
Comment (MariekeSpanFurore) [reply input/fsh/Profiles clinical datapoints/TumorFactors.fsh:287]: Shall I put this on the Jira as something to add later on when we have finished all the profiles?

Comment (vadi2) [reply input/fsh/Profiles clinical datapoints/TumorFactors.fsh:294]: Yes please

File: input/fsh/Profiles clinical datapoints/TumorFactors.fsh (hunk 13 above)
[the remark this answers is not in the document]
Comment (RicardoQuintano) [reply input/fsh/Profiles clinical datapoints/TumorFactors.fsh:91]: I understand. In this case, shouldn't these observations have different SCT codes?

**Comment by vadi2** [comment]
@MariekeSpanFurore feel free to merge this one when you feel it is ready.
